torch/_refs/__init__.py

```diff
@@ -1522,10 +1533,25 @@ def minimum(a: TensorLikeType, b: TensorLikeType) -> TensorLikeType:
     return prims.minimum(a, b)
 
 
+def shape_preserving_mul_custom(*args, **kwargs):
+    if isinstance(args[1], (int, float, SymInt, SymFloat)):
+        return args[0]
+    if isinstance(args[0], (int, float, SymInt, SymFloat)):
+        return args[1]
+    if args[0].shape == args[1].shape:
```

Comment:
that's closer to what's needed for binary outs, but still not quite (0d cpu + 0d cuda or 0d cuda + 0d cpu would both produce cuda output)

Reply:
sg